webapp/channels/src/components/admin_console/admin_console.tsx

```diff
@@ -49,39 +46,55 @@ type ExtraProps = {
     isCurrentUserSystemAdmin: boolean;
 }
 
-class AdminConsole extends React.PureComponent<Props, State> {
-    public constructor(props: Props) {
-        super(props);
-        this.state = {
-            search: '',
-        };
-    }
-
-    public componentDidMount(): void {
-        this.props.actions.getConfig();
-        this.props.actions.getEnvironmentConfig();
-        this.props.actions.loadRolesIfNeeded(['channel_user', 'team_user', 'system_user', 'channel_admin', 'team_admin', 'system_admin', 'system_user_manager', 'system_custom_group_admin', 'system_read_only_admin', 'system_manager']);
-        this.props.actions.selectLhsItem(LhsItemType.None);
-        this.props.actions.selectTeam('');
+const AdminConsole = (props: Props) => {
+    const [search, setSearch] = useState('');
+    const {hash} = props.location;
+    const lastFocusedLocation = useRef<Location>();
+
+    useEffect(() => {
+        props.actions.getConfig();
+        props.actions.getEnvironmentConfig();
+        props.actions.loadRolesIfNeeded(['channel_user', 'team_user', 'system_user', 'channel_admin', 'team_admin', 'system_admin', 'system_user_manager', 'system_custom_group_admin', 'system_read_only_admin', 'system_manager']);
+        props.actions.selectLhsItem(LhsItemType.None);
+        props.actions.selectTeam('');
         document.body.classList.add('console__body');
         document.getElementById('root')?.classList.add('console__root');
         resetTheme();
-    }
 
-    public componentWillUnmount(): void {
-        document.body.classList.remove('console__body');
-        document.getElementById('root')?.classList.remove('console__root');
-        applyTheme(this.props.currentTheme);
+        return () => {
+            document.body.classList.remove('console__body');
+            document.getElementById('root')?.classList.remove('console__root');
+            applyTheme(props.currentTheme);
 
-        // Reset the admin console users management table properties
-        this.props.actions.setAdminConsoleUsersManagementTableProperties();
-    }
+            // Reset the admin console users management table properties
+            props.actions.setAdminConsoleUsersManagementTableProperties();
+        };
+    }, []);
+
+    useLayoutEffect(() => {
```

Comment:
Can we add a comment explaining how to use this and what it does?

Reply:
moved it to a custom hook and added some comments